crates/matrix-sdk-base/src/client.rs

```diff
@@ -733,10 +725,13 @@ impl BaseClient {
         #[cfg(feature = "e2e-encryption")]
         let to_device = self
             .preprocess_to_device_events(
-                response.to_device.events,
-                &response.device_lists,
-                &response.device_one_time_keys_count,
-                response.device_unused_fallback_key_types.as_deref(),
+                EncryptionSyncChanges {
+                    to_device_events: response.to_device.events,
+                    changed_devices: &response.device_lists,
+                    one_time_keys_counts: &response.device_one_time_keys_count,
+                    unused_fallback_keys: response.device_unused_fallback_key_types.as_deref(),
+                    next_batch_token: None,
```

Comment:
We can just put `response.next_batch` in here. It is the correct token, though not really that useful since the usage of it from `/keys/query` disappeared from the spec. It would get rid of the optional though.

Reply:
Cool, I'll use `response.next_batch` here, but I'm nervous about making the `next_batch_token` optional; in the SS response, the `ToDevice` subsection might be empty, and I don't know in which cases it'll be empty (in particular, is it empty if there's no to-device events, despite us subscribing to to-device events?). I'll keep the option out of caution.